addons/account/models/account_move.py

```diff
@@ -1230,13 +1230,15 @@ def _compute_made_sequence_hole(self):
         self.env.cr.execute("""
             SELECT this.id
               FROM account_move this
+              JOIN res_company company ON company.id = this.company_id
          LEFT JOIN account_move other ON this.journal_id = other.journal_id
                                      AND this.sequence_prefix = other.sequence_prefix
                                      AND this.sequence_number = other.sequence_number + 1
              WHERE other.id IS NULL
                AND this.sequence_number != 1
                AND this.name != '/'
                AND this.id = ANY(%(move_ids)s)
+               AND (company.fiscalyear_lock_date IS NULL OR this.date >= company.fiscalyear_lock_date)
```

Comment:
*note for future readers:*
Not sure this is really useful to filter out here since it isn't annoying on the dashboard.
But on the other hand it might help the query planner to avoid doing the JOIN when browsing older data, when nothing can be done about the warning anymore.